### DISCUSSION TITLE
Add Headers to config_list

### DISCUSSION THREAD

**Discussion by llmdbboss** [microsoft/autogen#2422] [Q&A] [answered]
Is there a way to add headers like "Ocp-Apim-Subscription-Key" to the config_list?
I want to use a subscription key as mentioned in https://learn.microsoft.com/en-us/azure/api-management/api-management-subscriptions. 
Thanks!

**Comment by sonichi** [comment]
Have you tried adding "extra_headers" in the dict?

**Reply by llmdbboss** [reply]
Yes, i have tried both extra_headers and default_headers. But with both i get "Connection error."
For reference, this is what i am hoping to do:
config_list = [
        {
        "api_version" : "...",
        "base_url" : "...", 
        "model" : "....",
        "api_type" : "...",
        "api_key": "...",
        "extra_headers": {"Ocp-Apim-Subscription-Key": "..."}
        }
    ]

**Reply by sonichi** [reply]
Does using openai client directly work?

**Reply by llmdbboss** [reply]
I have put my open ai service behind Azure API Management gateway, so the OpenAI service has to be accessed by using the gateway URL.
I know that AzureChatOpenAI allows us to do this:
llm = AzureChatOpenAI(
    default_headers={"Ocp-Apim-Subscription-Key": <your-subscription-key-here>},
    deployment_name="gpt-35-turbo",
    model_name="gpt-35-turbo",
    temperature=0.1
)

**Reply by sonichi** [reply]
If it works for AzureOpenAI, the same should work for OpenAIWrapper in autogen.

**Reply by llmdbboss** [reply]
Any thoughts on why I am getting a connection error ?

**Comment by easontsai** [comment, marked as answer]
It work for me

config_list = [
            {
                "model": "gpt-4",
                "azure_deployment": "gpt-4",
                "api_key": "fake",
                "azure_ad_token": "Bearer ....",
                "base_url": "endpoint....",
                "api_type": "azure",
                "api_version": "api version....",
                "default_headers": {"Ocp-Apim-Subscription-Key": "secert.."},
            }
        ]

**Reply by llmdbboss** [reply]
I was missing the azure_ad_token. Thank you so much!

**Reply by Brutal15229** [reply]
what if it is password protected then how should proceed..? Because even with all this i am still not able to authorize as there is username and password as well
